server/jdkhttp-server/src/test/scala/sttp/tapir/server/jdkhttp/internal/KMPMatcherTest.scala

```diff
@@ -0,0 +1,49 @@
+package sttp.tapir.server.jdkhttp.internal
+
+import cats.effect.{IO, Resource}
+import org.scalatest.matchers.should.Matchers
+import sttp.tapir.server.jdkhttp.internal.KMPMatcher.{Match, NotMatched}
+import sttp.tapir.tests.{Test, TestSuite}
+
+import scala.concurrent.Future
+
+class KMPMatcherTest extends TestSuite with Matchers {
+
+  override def tests: Resource[IO, List[Test]] = Resource.eval(
+    IO.pure(
+      List(
+        Test("That matching over a set of bytes works and does not allow writing of any bytes if only matching") {
+          Future {
```

Comment:
Is this the correct way to add individual tests that has nothing to do with the servers themselves?
Felt a bit weird, but I'm not familiar enough to be sure.

Reply:
the `TestSuite` is more aimed at writing server/client interpreter tests. For normal unit tests, you can simply use what ScalaTest offers, see e.g. https://github.com/softwaremill/tapir/blob/f4b4d6bd37e8811d8285fdf0fa7dfdd9921289b1/core/src/test/scala/sttp/tapir/CodecTest.scala#L21

Reply:
Aha, makes sense. I'll fix :+1: